### PR TITLE
feat: MLIBZ-2333 Add method for setting instanceID

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
Add method for setting instance

#### Changes
Added method for setting instanceID to Client.Builder. 

#### Tests
Instrumented

### CODE REVIEW

File: android-lib/src/main/java/com/kinvey/android/Client.java [hunk 1]
```diff
@@ -814,7 +824,9 @@ public Client build(){
             if(this.MICBaseURL != null){
                client.setMICHostName(this.MICBaseURL);
             }
-
+            if (!Strings.isNullOrEmpty(this.instanceID)) {
+                client.setMICHostName(Constants.PROTOCOL_HTTPS + instanceID + Constants.HYPHEN + Constants.HOSTNAME_AUTH);
+            }
```
Comment (vinaygahlawat) [inline android-lib/src/main/java/com/kinvey/android/Client.java]: Will this construct a URL that does not end with a "/"? If so, is this consistent with how the default URL is formatted?

File: java-api-core/src/com/kinvey/java/AbstractClient.java [hunk 2]
```diff
@@ -463,6 +464,12 @@ public Builder setRequestTimeout(int requestTimeout) {
             return this;
         }
 
+        public Builder setInstanceID(String instanceID) {
+            this.instanceID = instanceID;
+            setBaseUrl(Constants.PROTOCOL_HTTPS + instanceID + Constants.HYPHEN + Constants.HOSTNAME_API);
+            return this;
```
Comment (vinaygahlawat) [inline java-api-core/src/com/kinvey/java/AbstractClient.java]: Same question as above regarding URL format.

File: android-lib/src/main/java/com/kinvey/android/Client.java (hunk 1 above)
Comment (yuliya-guseva) [reply android-lib/src/main/java/com/kinvey/android/Client.java]: The SDK has logic for adding  "/" to the end of url. So all URLs will have "/" at the end.
https://github.com/Kinvey/java-library/blob/7c8e85107593870941a39c249e93464bc6238ff3/java-api-core/src/com/kinvey/java/AbstractClient.java#L115
`this.MICHostName = MICHostName.endsWith("/") ? MICHostName : MICHostName + "/"`

File: java-api-core/src/com/kinvey/java/AbstractClient.java (hunk 2 above)
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/AbstractClient.java]: `setBaseUrl` also has logic for adding "/" to the end of url.

**Pull request merged**